services/motion/builtin/builtin.go

```diff
@@ -274,22 +274,44 @@ func (ms *builtIn) MoveOnGlobe(
 	if err != nil {
 		return false, err
 	}
+	currentPosition := currentPIF.Pose().Point()
+
+	// get position of localizer relative to base
+	robotFS, err := ms.fsService.FrameSystem(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// build maps of relevant components and inputs from initial inputs
+	fsInputs, _, err := ms.fsService.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	localizerFrame := robotFS.Frame(movementSensorName.ShortName())
+	if localizerFrame != nil {
+		tf, err := robotFS.Transform(fsInputs, &currentPIF, movementSensorName.Name)
```

Comment:
I think I'm missing something here.
`currentPIF` is returned by the movement sensor, so should be in the frame of the movement sensor. Isn't this transforming a pose to its own reference frame? Shouldn't `componentName` be used at some point?

Reply:
Agree, I don't think this does anything and we should either be transforming to the Base frame

Reply:
when we use `movementSensorName`, inside the `Transform` function the value of `tfParent.pose.Point()` is what is specified within the frame section for movement_sensor of the passed in config.
`Transform` ends with `return object.Transform(tfParent), nil` so it makes sense to transform `object` which is `currentPIF` by the translation of the movement sensor's frame which is stored in `tfParent`.

Reply:
I'm not following; where do you see `tfParent.pose.Point()`?

What the `framesystem.Transform` function does, is say "Here is pose A, in frame B. What would pose A be if we observe it from frame C instead of from frame B?"

As far as I can tell, both frame B and frame C in this code are the movement sensor. You're not transforming `object *by* the movement sensor's transform, you're transforming it *to* the movement sensor's frame of reference, which is where it already is.